integration-tests/capabilities/environment-cache.toml

```diff
@@ -0,0 +1,233 @@
+[blockchain_a]
+type = 'anvil'
+image = 'f4hrenh9it/foundry:latest'
+pull_image = false
+port = '8545'
+port_ws = ''
+chain_id = '1337'
+docker_cmd_params = ['-b', '5']
+public_key = ''
+contracts_dir = ''
+
+[blockchain_a.out]
+use_cache = true
+family = 'evm'
+container_name = 'blockchain-node-ba2f0'
+chain_id = '1337'
+
+[[blockchain_a.out.nodes]]
+ws_url = 'ws://127.0.0.1:8545'
+http_url = 'http://127.0.0.1:8545'
+docker_internal_ws_url = 'ws://blockchain-node-ba2f0:8545'
+docker_internal_http_url = 'http://blockchain-node-ba2f0:8545'
+
+[nodeset]
+nodes = 5
+http_port_range_start = 0
+p2p_port_range_start = 0
+dlv_port_range_start = 0
+override_mode = 'each'
+
+[nodeset.db]
+image = 'postgres:15.6'
+port = 0
+name = ''
+volume_name = ''
+databases = 5
+jd_database = false
+pull_image = false
+
+[[nodeset.node_specs]]
+[nodeset.node_specs.node]
+image = ''
+name = ''
+docker_file = 'plugins/chainlink.Dockerfile'
+docker_ctx = '../..'
+pull_image = false
+capabilities = ['./streams-linux-amd64', './cron-linux-amd64']
+capabilities_container_dir = ''
+test_config_overrides = "\n\t\t\t\t[Feature]\n\t\t\t\tLogPoller = true\n\n\t\t\t\t[OCR2]\n\t\t\t\tEnabled = true\n\t\t\t\tDatabaseTimeout = '1s'\n\n\t\t\t\t[P2P.V2]\n\t\t\t\tEnabled = true\n\t\t\t\tListenAddresses = ['0.0.0.0:5001']\n\t\t\t\tDefaultBootstrappers = ['12D3KooWHnPLxdrpqCfTuy98sk6ZbN1aRwWYZWRoaJaHTppYJpxn@localhost:5001']\n\n\t\t\t\t[Capabilities.Peering.V2]\n\t\t\t\tEnabled = true\n\t\t\t\tListenAddresses = ['0.0.0.0:6690']\n\t\t\t\tDefaultBootstrappers = ['12D3KooWHnPLxdrpqCfTuy98sk6ZbN1aRwWYZWRoaJaHTppYJpxn@localhost:6690']\n\n\t\t\t\t# This is needed for the target capability to be initialized\n\t\t\t\t[[EVM]]\n\t\t\t\tChainID = '1337'\n\n\t\t\t\t[[EVM.Nodes]]\n\t\t\t\tName = 'anvil'\n\t\t\t\tWSURL = 'ws://blockchain-node-ba2f0:8545'\n\t\t\t\tHTTPURL = 'http://blockchain-node-ba2f0:8545'\n\n\t\t\t\t[EVM.Workflow]\n\t\t\t\tFromAddress = '0x7095A8B3eac565787b993973A0B303C56564a196'\n\t\t\t\tForwarderAddress = '0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9'\n\t\t\t\tGasLimitDefault = 400_000\n\n\t\t\t\t[Capabilities.ExternalRegistry]\n\t\t\t\tAddress = '0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512'\n\t\t\t\tNetworkID = 'evm'\n\t\t\t\tChainID = '1337'\n\n\t\t\t\t[Capabilities.WorkflowRegistry]\n\t\t\t\tAddress = \"0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9\"\n\t\t\t\tNetworkID = \"evm\"\n\t\t\t\tChainID = \"1337\"\n\n\t\t\t\t[Capabilities.GatewayConnector]\n\t\t\t\tDonID = \"1\"\n\t\t\t\tChainIDForNodeKey = \"1337\"\n\t\t\t\tNodeAddress = '0x7095A8B3eac565787b993973A0B303C56564a196'\n\n\t\t\t\t[[Capabilities.GatewayConnector.Gateways]]\n\t\t\t\tId = \"por_gateway\"\n\t\t\t\tURL = \"ws://node4:5003/node\"\n\t\t\t"
```

Comment:
If I understand this config correctly, all nodes are listening on 0.0.0.0:5001 AND on 0.0.0.0:6690 -- is that correct?

If so, that seems like a misconfiguration to me

The correct setup is that:
- each node has its own listen address for both P2P and capabilities; these shouldn't clash
- each node sets a default bootstrapper to talk to for both P2P and capabilities; they also shouldn't clash

For OCR you need to configure the bootstrapper by configuring a bootstrap job; for capabilities the bootstrap job is configured automatically when the node sees its own P2P Id in the list of DefaultBootstrappers

Reply:
(All of the above is assuming that they're sharing the same network ofc)